pkg/cli/repositories.go

```diff
@@ -0,0 +1,447 @@
+
+		// Check if file exists
+		if !flagForce {
+			// if stat returns no error return error saying you need to provide the force flag
+			_, err := os.Stat(filepath.Join(pathCheck, name))
+			if err == nil {
+				SetError(errors.New("key file already exists provide --force to overwrite"), 2)
+				return
+			}
+		}
+
+		// Write key to a file under that keys directory
+		err = ioutil.WriteFile(filepath.Join(pathCheck, name), []byte(key), os.ModePerm)
+		if err != nil {
+			SetError(err, 3)
+			return
+		}
+
+		// Check if default and write another file called default under repository-keys
+		if flagDefault {
+			err = ioutil.WriteFile(filepath.Join(pathCheck, "default"), []byte(key), os.ModePerm)
+			if err != nil {
+				SetError(err, 4)
+				return
+			}
+		}
+	},
+}
+
+func init() {
+	f := createKeyCmd.Flags()
+	f.BoolVar(&flagDefault, "default", false, "save this key to use as default")
+	f.BoolVar(&flagForce, "force", false, "force overwrite of key file")
+}
+
+var listKeysCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List all keys currently stored",
+	Args:  cobra.MaximumNArgs(0),
+	Run: func(cmd *cobra.Command, args []string) {
+		usr, err := os.UserHomeDir()
+		if err != nil {
+			SetError(err, 1)
+			return
+		}
+		pathCheck := filepath.Join(usr, ".vorteil", "repository-keys")
+
+		if _, err = os.Stat(pathCheck); os.IsNotExist(err) {
+			fmt.Printf("no keys found\n")
+			return
+		}
+
+		fis, err := ioutil.ReadDir(pathCheck)
+		if err != nil {
+			SetError(err, 2)
+			return
+		}
+
+		for _, fi := range fis {
+			if fi.Name() != "default" {
+				fmt.Println(fi.Name())
+			}
+		}
+	},
+}
+
+var deleteKeyCmd = &cobra.Command{
+	Use:   "delete NAME",
+	Short: "Delete a key currently stored",
+	Args:  cobra.MaximumNArgs(1),
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			return errors.New("Must provide the name of the key you want to delete")
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		name := args[0]
+
+		usr, err := os.UserHomeDir()
+		if err != nil {
+			SetError(err, 1)
+			return
+		}
+		pathCheck := filepath.Join(usr, ".vorteil", "repository-keys")
+		path := filepath.Join(pathCheck, name)
+		dpath := filepath.Join(pathCheck, "default")
+
+		// before removing we should check if default is the same and delete that
+		f1, err := ioutil.ReadFile(path)
+		if err != nil {
+			SetError(fmt.Errorf("%s keyfile does not exist", name), 2)
+			return
+		}
+
+		f2, err := ioutil.ReadFile(dpath)
+		if err != nil {
+			if !errors.Is(err, os.ErrNotExist) {
+				SetError(err, 3)
+				return
+			}
+		}
+
+		// If same bytes remove default aswell
+		if bytes.Equal(f1, f2) {
+			err = os.Remove(dpath)
+			if err != nil {
+				SetError(err, 4)
+				return
+			}
+		}
+
+		// Else just remove the keyfile
+		err = os.Remove(filepath.Join(pathCheck, name))
+		if err != nil {
+			SetError(err, 5)
+			return
+		}
+
+	},
+}
+
+var pushCmd = &cobra.Command{
```

Comment:
I cant seem to push at all, errors I've seen so far.

**Pushing project to existing app**
```sh
$ ./borteil repositories push http://devimg.vorteil.io jon.alfaro/beee/161456131 /tmp/zzz
Preparing Package  ⠋                                                                                
Uploading Package   1s [=======================>------------------------------------------------------] 1.2 MiB / 4.0 MiB
Post "http://devimg.vorteil.io/organisations/jon.alfaro/buckets/beee/apps/161456131": write tcp 192.168.1.60:60322->34.96.74.226:80: write: broken pipe
```

**Pushing project to existing app 2**
```sh
$ ./borteil repositories push http://devimg.vorteil.io jon.alfaro/beee/161456131 /tmp/zzz/
Preparing Package  ⠋                                                                                
Uploading Package   1s [===============>--------------------------------------------------------------] 864.0 KiB / 4.0 MiB
```

**Pushing package to new app**
```sh
$ ./borteil repositories push http://devimg.vorteil.io jon.alfaro/beee/yeet ~/Downloads/helloworld.vorteil 
Preparing Package  ⠋                                                                                
Uploading Package   3s [========>---------------------------------------------------------------------] 480.0 KiB / 4.0 MiB
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x11420e6]

goroutine 51 [running]:
github.com/vbauerster/mpb/v5.(*Progress).Wait(0x0)
	/home/jon/go/pkg/mod/github.com/vbauerster/mpb/v5@v5.3.0/progress.go:186 +0x26
github.com/vorteil/vorteil/pkg/elog.(*pb).Finish(0xc00034c3c0, 0xc000260000)
	/home/jon/go/src/github/vorteil/vorteil/pkg/elog/logger.go:272 +0x125
github.com/vorteil/vorteil/pkg/elog.(*pb).ProxyReader.func2(0xc00013a140, 0x25ba3e0)
	/home/jon/go/src/github/vorteil/vorteil/pkg/elog/logger.go:324 +0x65
github.com/vorteil/vorteil/pkg/vio.(*lazyReadCloser).Close(0xc00078c3c0, 0x25ba3e0, 0xc0008f0040)
	/home/jon/go/src/github/vorteil/vorteil/pkg/vio/file.go:270 +0x7b
net/http.(*Request).closeBody(...)
	/home/jon/go/src/net/http/request.go:1388
net/http.(*persistConn).writeLoop(0xc0000d46c0)
	/home/jon/go/src/net/http/transport.go:2359 +0x399
created by net/http.(*Transport).dialConn
	/home/jon/go/src/net/http/transport.go:1709 +0xcdc
```

Also `$ echo $?` is returning `0`

Seems Uploading Package prematurely finishes.

Reply:
Setting target repo to image-metadata-service gives a different error
```sh
$ ./borteil repositories push http://devimg.vorteil.io/metadata jon.alfaro/beee/zzzz ~/Downloads/helloworld.vorteil 
Preparing Package  ⠋                                                                                
Uploading Package   0s [=====================>--------------------------------------------------------] 1.2 MiB / 4.0 MiB
500 Internal Server Error: cannot convert 14 to Int8
```